pixeltable/functions/bfl.py

```diff
@@ -0,0 +1,512 @@
+                        raise BflUnexpectedError(f'BFL polling error (status {resp.status}): {error_text}')
+
+        raise BflUnexpectedError(f'BFL task {task_id} timed out after {max_wait} seconds')
+
+    async def _download_image(self, task_id: str, url: str) -> PIL.Image.Image:
+        """Download image from the result URL."""
+        async with self.session.get(url) as resp:
+            if resp.status != 200:
+                raise BflUnexpectedError(f'BFL task {task_id}: failed to download image, status {resp.status}')
+            img_data = await resp.read()
+            if len(img_data) == 0:
+                raise BflUnexpectedError(f'BFL task {task_id} resulted in an empty image')
+            img = PIL.Image.open(BytesIO(img_data))
+            img.load()
+            _logger.debug(
+                f'BFL task {task_id} successful. Image bytes: {len(img_data)}, size: {img.size}'
+                f', format: {img.format}, mode: {img.mode}'
+            )
+            return img
+
+    async def generate(self, endpoint: str, payload: dict) -> PIL.Image.Image:
+        """Submit a generation task and wait for the result."""
+        task_id, polling_url = await self._submit_task(endpoint, payload=payload)
+        _logger.debug(f'BFL task {task_id} submitted to {endpoint}')
+        return await self._poll_result(task_id, polling_url)
+
+
+@register_client('bfl')
+def _(api_key: str) -> _BflClient:
+    return _BflClient(api_key=api_key)
+
+
+def _client() -> _BflClient:
+    return Env.get().get_client('bfl')
+
+
+# Model endpoint mapping
+_MODEL_ENDPOINTS = {
+    # FLUX.2 models
+    'flux-2-pro': '/v1/flux-2-pro',
+    'flux-2-flex': '/v1/flux-2-flex',
+    'flux-2-max': '/v1/flux-2-max',
+    # FLUX 1.1 models
+    'flux-pro-1.1': '/v1/flux-pro-1.1',
+    'flux-pro-1.1-ultra': '/v1/flux-pro-1.1-ultra',
+    # FLUX.1 dev
+    'flux-dev': '/v1/flux-dev',
+    # Kontext models (for editing)
+    'flux-kontext-pro': '/v1/flux-kontext-pro',
+    'flux-kontext-max': '/v1/flux-kontext-max',
+}
+
+
+@pxt.udf(resource_pool='request-rate:bfl')
+async def generate(
+    prompt: str,
+    *,
+    model: Literal[
+        'flux-2-pro', 'flux-2-flex', 'flux-2-max', 'flux-pro-1.1', 'flux-pro-1.1-ultra', 'flux-dev'
+    ] = 'flux-2-pro',
+    width: int | None = None,
+    height: int | None = None,
+    seed: int | None = None,
+    safety_tolerance: int | None = None,
+    output_format: Literal['jpeg', 'png'] | None = None,
+    steps: int | None = None,
+    guidance: float | None = None,
+) -> PIL.Image.Image:
+    """
+    Generate an image from a text prompt using FLUX models.
+
+    This UDF wraps the BFL FLUX API endpoints. For more information, refer to the official
+    [API documentation](https://docs.bfl.ai/flux_2/flux2_text_to_image).
+
+    Args:
+        prompt: Text description of the image to generate.
+        model: FLUX model to use. Options:
+            - 'flux-2-pro': Best balance of speed and quality (default)
```

Comment:
Incorrect markdown (bulleted lists must be preceded by a blank line)

But I'd argue against hard-coding a list of models in our docs. We don't do that for any other provider - unless there's a specific reason why BFL should be treated differently in this regard.

Just link to the models section of the BFL docs and let them do the work of keeping it up to date.